lib/elixir/lib/enum.ex

```diff
@@ -2537,16 +2580,60 @@ defmodule Enum do
 
   @compile {:inline, enum_to_string: 1}
 
-  defp enumerate_and_count(enumerable, count) when is_list(enumerable) do
-    {enumerable, length(enumerable) - abs(count)}
+  defp enum_to_string(entry) when is_binary(entry), do: entry
+  defp enum_to_string(entry), do: String.Chars.to_string(entry)
+
+  defp enumerate_count_index(enumerable, index) when is_list(enumerable) do
+    length = length(enumerable)
+    case normalize_index(index, length) do
+      i when is_integer(i) and i < length ->
+        {:ok, enumerable, length, i}
+      _ ->
+        :error
+    end
+  end
+
+  defp enumerate_count_index(enumerable, index) do
+    case get_size_index(enumerable, index) do
```

Comment:
I don't understand why you call `get_size_index` here. It seems that every time you call `enumerate_count_index` you already tried to call `get_size_index`, so this one will always return `{:error, module}`?